ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -444,6 +419,15 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr = do
     assert (isJust $ lookupBlockInfo (headerHash hdr)) $ return ()
 
     if
+      -- If the block is already in the current chain, ignore it. As we add
+      -- blocks to the VolatileDB before adding them to the chain selection
+      -- queue, it can often happen that when we did chain selection for a
+      -- block, we already extended its candidate chain with its successors
+      -- (that are in the queue) as they are already part of the VolatileDB.
+      | AF.withinFragmentBounds (blockPoint hdr) curChain -> do
+        trace $ IgnoreBlockAlreadyInChain p
+        return tipPoint
```

Comment:
Without this case, this PR would significantly slow down sync speeds (certainly for Byron), as this happens very often, resulting in the expensive `TrySwitchToAFork` case.